src/lib/common/sol-platform-impl-systemd.c

```diff
@@ -743,29 +774,27 @@ int
 sol_platform_register_locale_monitor(void)
 {
     sd_bus *bus;
+    int r;
 
-    if (_ctx.locale)
+    if (_ctx.timedate_monitor_registered)
         return 0;
 
-    bus = sol_bus_get(NULL);
-    SOL_NULL_CHECK(bus, -ENOTCONN);
-
-    _ctx.locale = sol_bus_client_new(bus, "org.freedesktop.locale1");
-    SOL_NULL_CHECK_GOTO(_ctx.locale, error);
+    bus = _get_sd_bus(_ctx.locale, _localed_bus_initialized);
+    SOL_NULL_CHECK(bus, -EINVAL);
 
-    return sol_bus_map_cached_properties(_ctx.locale, "/org/freedesktop/locale1",
+    r = sol_bus_map_cached_properties(_ctx.locale, "/org/freedesktop/locale1",
```

Comment:
[off-topic] I wish I changed this API to return a handle, that I would pass to `_unmap_cached_properties()` :-/